orc8r/cloud/go/obsidian/access/rest_middleware.go

```diff
@@ -125,15 +125,24 @@ func TokenMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
 			return echo.NewHTTPError(http.StatusBadRequest, "failed to parse basic auth header")
 		}
 
-		getPDReq := &certprotos.GetPolicyDecisionRequest{
+		resourceType, resourceVal := getResource(c)
+		getPDReq := certprotos.GetPolicyDecisionRequest{
 			Username: username,
 			Token:    token,
 			Resource: &certprotos.Resource{
-				Action:   getRequestAction(req, nil),
-				Resource: req.RequestURI,
+				Action:       getRequestAction(req, nil),
+				ResourceType: resourceType,
+				Resource:     req.RequestURI,
 			},
```

Comment:
I think there's still a slight disconnect on what we wanna send here. Specifically, we always want to send *both* the request URI and and network ID (if any)

So e.g.

```go
		getPDReq := certprotos.GetPolicyDecisionRequest{
			Username: username,
			Token:    token,
			Resource: &certprotos.Resource{
				Action:       getRequestAction(req, nil),
				Resource:   req.RequestURI,
				NetworkID: /* network_id, if any */,
			},
```

That way, whether the token contains a path-based vs. networkID-based vs. tenantID-based token, it has all the info it needs to make a decision

Reply:
I set it in the next few lines, bc using one of in proto makes it so difficult to just put any field.